fieldpath/element.go

```diff
@@ -70,6 +71,24 @@ func (e PathElement) String() string {
 	}
 }
 
+// Follow tries tries to find the value corresponding to the
+// PathElement. If it can't follow the PathElement, an error is
+// returned.
+func (e PathElement) Follow(v value.Value) (value.Value, error) {
+	switch {
+	case e.FieldName != nil:
+		return v.Map.Follow(*e.FieldName)
```

Comment:
I'm not sure about this--
* Do we need this function on untyped values?
* Is it good to add a follow function to every type? (e.g., why not just call get here)
* It seems you'll need to verify the type of the value before blindly accessing something in the one-of
* ...so maybe even if we need this for untyped values, it should be implemented in the typed package?

Reply:
> Do we need this function on untyped values?

Hard to tell. Right now I mostly need it to get the version of the object.

> Is it good to add a follow function to every type? (e.g., why not just call get here)

I could have done it, this function would have been more complicated, while I can spread the complexity to each individual type (better encapsulation?). Not super strong.

> It seems you'll need to verify the type of the value before blindly accessing something in the one-of

This is still checking if it works or not. I'm checking that we do have a map when we're requesting a field etc. Also if we are acting on typed object, we maybe can assume that it's been validated before.

One of my problem is that I'd like this to work on value and typedvalue, and when we're using typedvalue, it'd be nice if the type information would come with it.

Reply:
I'm still thinking it'd be better to put this in the typed package; it can still work on untyped objects (we can allow it to go inside untyped blobs).

Reply:
I think the thing that mostly bothers me is that this functionality crosses package boundaries.